File: WordApp/Models/AnsweredWordModel.cs
using WordApp.Models.Base;

namespace WordApp.Models
{
    public class AnsweredWordModel: BaseModel
    {
        public virtual AssignableWordTaskModel AssignedWordTask { get; set; }
        public virtual AnswerModel Answer { get; set; }
        public virtual WordModel Word { get; set; }
    }
}

File: WordApp/ClientApp/src/app/tasks/models/answered-word.model.ts
import { WordModel } from './../../models/words/word.model';
import { TaskAnswerModel } from './task-answer.model';

export class AnsweredWordModel {
  word: WordModel;
  answer: TaskAnswerModel;

  constructor(word: WordModel, answer: TaskAnswerModel) {
    this.word = word;
    this.answer = answer;
  }
}
